src/utils/resolveIdViaPlugins.ts

```diff
@@ -24,8 +24,24 @@ export function resolveIdViaPlugins(
 		}
 		replaceContext = (pluginContext, plugin): PluginContext => ({
 			...pluginContext,
+			// @ts-expect-error returns null instead of Promise.resolve(null) by design
```

Comment:
If you return null, wouldn't this defeat the assumption that this.resolve returns a Promise? This would breaks plugins that append a .then(...) to this.resolve. The link you added below is about the `resolveId` plugin hook, and not the `this.resolve` context function. Also, I definitely do not like `@ts-expect-error` as it completely turns off type checking and easily hides legitimate errors.

Reply:
Thanks for the review. `@ts-expect-error` is definitely not a good way. Previously I was misled by the doc, because it says **"If null is returned, the import could not be resolved by Rollup or any plugin but was not explicitly marked as external by the user. "**. I think doc is outdated, and changes are included in this PR.